src/librustc_session/lint/builtin.rs

```diff
@@ -532,6 +532,12 @@ declare_lint! {
     "unsafe operations in unsafe functions without an explicit unsafe block are deprecated",
 }
 
+declare_lint! {
+    pub ASM_UNUSED_ARGUMENT,
```

Comment:
```suggestion
    pub UNUSED_ASM_ARGUMENTS,
```
We have some [lint naming conventions](https://rust-lang.github.io/rfcs/0344-conventions-galore.html#lints).